File: 프로그래머스/2/131127. 할인 행사/할인 행사.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string[] want, int[] number, string[] discount) {
        int answer = 0;
        int len = discount.Length;
        int total = len - 10 + 1;
        
        var baseDB = new Dictionary<string, int>();
        for(int i = 0; i < want.Length; i++)
        {
            baseDB[want[i]] = number[i];
        }
        
        for(int i = 0; i < total; i++)
        {
            var countDB = new Dictionary<string, int>();
            for(int j = 0; j < 10; j++)
            {
                if(!countDB.ContainsKey(discount[i + j]))
                {
                    countDB[discount[i + j]] = 1;
                }
                else
                {
                    countDB[discount[i + j]]++;
                }
            }
            
            bool isOK = true;
            
            foreach(var kv in baseDB)
            {
                if(!countDB.ContainsKey(kv.Key) || countDB[kv.Key] != kv.Value)
                {
                    isOK = false;
                    break;
                }
            }
            if(isOK) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    for(int start = 0; start <= discount.size() - 10; start++)
    {
        map<string, int> mp;
        
        for(int i = start; i < start + 10; i++)
        {
            mp[discount[i]]++;
        }
        
        bool ok = true;
        for(int i = 0; i < want.size(); i++)
        {
            if(mp[want[i]] != number[i])
            {
                ok = false;
                break;
            }
        }
        if(ok) answer++;
    }
    return answer;
}